goosebit/updater/manager.py

```diff
@@ -170,16 +169,26 @@ async def get_update_file(self) -> FirmwareArtifact:
             if rollout and not rollout.paused:
                 file = rollout.fw_file
 
-        return FirmwareArtifact(file, device.hw_model, device.hw_revision)
+        if file == "latest":
+            return await FirmwareUpdate.latest()
+        if file == "pinned":
+            return await FirmwareUpdate.get(
+                uri=UPDATES_DIR.joinpath(device.fw_file).absolute().as_uri()
+            )
+
+        compat = FirmwareCompatibility.get(
+            hw_model=device.hw_model, hw_revision=device.hw_revision
+        )
+        return await FirmwareUpdate.get(
+            uri=UPDATES_DIR.joinpath(device.fw_file).absolute().as_uri(),
+            compatibility=compat,
+        )
 
     async def get_update_mode(self) -> str:
         device = await self.get_device()
 
         file = await self.get_update_file()
-        if file.is_empty():
```

Comment:
Why removing this case? At least when the system starts and devices connect there are no firmware available and no update should be rolled out.